### ISSUE TITLE
Use "=" as an optional separator of a long option and its value

### ISSUE THREAD

**Issue by heavywatal** [muellan/clipp#12]
Is it possible to use `=` as an option-value separator like in boost::program_options and [Python argparse](https://docs.python.org/3/library/argparse.html#option-value-syntax)?

Current behavior (AFAIK):
```
# (option("--longopt") & value("arg", target))
Command           target
--longopt foo  => "foo"
--longoptfoo   => "foo"
--longopt=foo  => "=foo"

# (option("--longopt=") & value("arg", target))
Command           target
--longopt foo  => Not match
--longoptfoo   => Not match
--longopt=foo  => "foo"
```

Proposed behavior:
```
Command           target
--longopt foo  => "foo"
--longopt=foo  => "foo"
```

Thanks.

**Comment by muellan** [comment]
Yes, if you write the option like this:
```option("--longopt", "--longopt=") & value("arg", target)```
the following input command line arguments will be allowed:
 - ```{"--longopt", "SOMEVALUE"}```
 - ```{"--longoptSOMEVALUE"}```
 - ```{"--longopt=", "SOMEVALUE"}```
 - ```{"--longopt=SOMEVALUE"}```

The parser is designed in a character-agnostic way and no character is given special treatment. You could use 漢字 for everything and the parser wouldn't care. If you have
```option("?opt", "+?opt", "?opt=") & value("arg", target)```
The only thing that the parser checks, is, that the input contains a (sequence of) command line argument(s) with a prefix in the set {"-opt", "--opt", "?opt="} and a suffix, that can be used as a value.

Note that the generated documentation will have a space in between the option string and value label:
```--longopt= arg```

**Comment by heavywatal** [comment]
Thank you for the prompt answer! Now I have learned how to write a program that accepts either with or without `=`. Here is another question. Can I hide a specific option flag from the doc? E.g., showing only `--longopt` from `option("--longopt", "--longopt=")`. I found `doc_formatting::max_flags_per_param_in_doc(1)` could work, but it is unfortunately a global option for the entire doc. Another potential solution may be to define two separate sets of `option & value` that have the same target variables: one with doc, the other without doc. Hopefully you know better ways to do that.
